### PR TITLE
Searchbox: Enter jumps to tracks if search query was transmitted

### PR DESCRIPTION
**Pull request by ronso0**
1st Enter press: emit query (or debounce timeout expired)
2nd Enter press: jump to tracks table

This is mainly supposed to be a shortcut for when working with the keyboard. Currently it requires 3 x Tab to get from searchbox to tracks, while it's just one press of the Browse/Trax encoder, often mapped to `[Library], GoToItem`.

I'm planning to add some visual feedback to signal that the query was submitted, it's not clear what that'll be though.
* could be a another pushbutton next to Clear, e.g. ➜, that's greyed out as soon as the search is triggered, though that's cumbersome to integrate (layout-wise, considering we also expect right-to-left locales)
* could be a 3rd state of the Clear icon (normal | focused | query submitted)
* ...

### CODE REVIEW

**Comment by daschuer** [comment]
This sounds like a useful feature. 

I have not tested yet, but I assume that the current search is replaced by the saved search in tracks after the switch. Is this the case? 

What do you think about the status widget idea below the search bar, with a button that applies the same search for all tracks?

**Comment by daschuer** [comment]
Ah, I see, I have misread the PR. Just tested and it works. I don't think that this feature requires a visual feedback. 

However my personal main issue is the missing workflow to search in the Tracks feature after browsing the library tree.

Can we merge this as it is?

**Comment by ronso0** [comment]
> Can we merge this as it is?

I adjusted the tooltip, so: yes!

File: src/widget/wsearchlineedit.cpp [hunk 1]
```diff
@@ -207,6 +208,8 @@ void WSearchLineEdit::setup(const QDomNode& node, const SkinContext& context) {
             tr("Shortcuts") + ": \n" +
             tr("Ctrl+F") + "  " +
             tr("Focus", "Give search bar input focus") + "\n" +
+            tr("Return") + " " +
+            tr("Trigger search, next press jumps to tracks view") + "\n" +
```
Comment (daschuer) [inline src/widget/wsearchlineedit.cpp:212]: Is this still correct? In my test the search is instantly applied.
And return immediately moves to the Tracks table.

**Comment by daschuer** [comment]
Please remove the draft state when done.

File: src/widget/wsearchlineedit.cpp (hunk 1 above)
Comment (daschuer) [reply src/widget/wsearchlineedit.cpp:212]: Ah, so it is the long debounce time that eats the first return.
How about this: 
```suggestion
            tr("Trigger search before search-as-you-type timeout or jumps to tracks view afterwards") + "\n" +
```

Comment (daschuer) [reply src/widget/wsearchlineedit.cpp:212]: Did you consider this comment?

Comment (ronso0) [reply src/widget/wsearchlineedit.cpp:212]: Oh, added it now. Thanks!

**Review by daschuer — APPROVED** [review]
LGTM, Thank you.

**Pull request merged**